microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsCodeLensParticipant.java

```diff
@@ -72,18 +76,22 @@ public boolean isAdaptedForCodeLens(JavaCodeLensContext context, IProgressMonito
 	public List<CodeLens> collectCodeLens(JavaCodeLensContext context, IProgressMonitor monitor) throws CoreException {
 		ITypeRoot typeRoot = context.getTypeRoot();
 		IJavaElement[] elements = typeRoot.getChildren();
-		int serverPort = JaxRsContext.getJaxRsContext(context).getServerPort();
+		JaxRsContext jaxrsContext = JaxRsContext.getJaxRsContext(context);
+		int serverPort = jaxrsContext.getServerPort();
 		IJDTUtils utils = context.getUtils();
 		MicroProfileJavaCodeLensParams params = context.getParams();
 		params.setLocalServerPort(serverPort);
 		List<CodeLens> lenses = new ArrayList<>();
-		collectURLCodeLenses(elements, null, lenses, params, utils, monitor);
+		collectURLCodeLenses(jaxrsContext, elements, null, lenses, params, utils, monitor);
 		return lenses;
 	}
 
-	private static void collectURLCodeLenses(IJavaElement[] elements, String rootPath, Collection<CodeLens> lenses,
+	private static void collectURLCodeLenses(JaxRsContext jaxrsContext, IJavaElement[] elements, String rootPath, Collection<CodeLens> lenses,

```

Comment:
Don't use params but only JaxRsContext:

private static void collectURLCodeLenses(IJavaElement[] elements, String rootPath, Collection<CodeLens> lenses,
			JaxRsContext context, IJDTUtils utils, IProgressMonitor monitor)

Reply:
By removing the params param, there is a lot of refactoring, where there isn't any other method/class that performs similar tasks (e.g. params.isCheckServerAvailable() and params.getOpenURICommand(), both of which can't be moved and shouldn't be copied). Do you still want to remove params here?

Reply:
Ok let's keep params only for  params.isCheckServerAvailable() and params.getOpenURICommand(). But we should have JaxRSContext#getLocalBaseURL().

Please move JaxRSContext (from the first parameter) after the params.